### ISSUE TITLE
How to create 2D psf？

### ISSUE THREAD

**Issue by Xuesm** [TristaZeng/ZS-DeconvNet#15]
Hello, thank you very much for sharing your excellent work. I have successfully trained a new model using the 2D data and PSF provided by you. I am now replacing it with my own 2D WF data, but I found that the creat_psf. m provided by you created a 3D PSF. Can you tell me how you create the 2D WF PSF? And does dxy refer to pixel size in creat_psf. m?  Looking forward your prompt reply!

**Comment by TristaZeng** [comment]
2D WF PSF is the axial center slice of the 3D PSF, you can set SizeZ=1. dxy is the lateral pixel size of PSF.

**Comment by Xuesm** [comment]
thanks for your reply，helped me a lot!

**Comment by Xuesm** [comment]
Hello, may I ask how dxy is calculated in the created PDF. m file you provided? I found that the values in the file are not quite what I understood. Is there any calculation relationship between it and the size of the detector focal plane pixel-size and the lens magnification? Looking forward to your reply, thank you very much!
![1](https://github.com/user-attachments/assets/3db417f7-f5f4-4298-b71a-aaa79288f6b5)

**Comment by TristaZeng** [comment]
The pixel size of the output PSF will be `dxy`, which you specified, and usually you specify it to be the actual sampling period (calculated according to the optical system, given the detector pixel size, magnification rate, etc) of the specimen. But in practice you can specify it to be small enough values to create a finer PSF, and enter the psf `dxy` value during training, and the training code will automatically interpolate the PSF to match with the data `dxy`.

**Comment by Xuesm** [comment]
Thank you very much for your reply! I found that the values of dx and dy here are the same as dxy. What do dx and dy mean？ How do you set these two values? Looking forward to your reply, thanks a lot!
![2](https://github.com/user-attachments/assets/ef797c4f-61ca-4acf-8eab-e9ea1c836c9d)

**Comment by TristaZeng** [comment]
dx and dy are usually the same. They are the horizontal and vertical sampling distance.

**Comment by Xuesm** [comment]
Got it!  Thanks bro! I will try again.

**Comment by Xuesm** [comment]
Hello, did you find ringing artifacts in the deconvoluted results? Is it due to inaccurate PSF? The following two images are the original image and deconved_iter50000 image of my test on ZS-DeconvNet.
![nirOri](https://github.com/user-attachments/assets/05130d3d-90fd-4809-a06e-bcf738bf0d86)
![nirDeconv](https://github.com/user-attachments/assets/55cba94e-a9ea-4534-93af-276133946d11)

**Comment by TristaZeng** [comment]
Yes it seems very much like inaccurate PSF. If you are upsampling, use /2 `dxy`.

**Comment by Xuesm** [comment]
Hello, sorry to bother you again. I don't understand what you mean by using/2 dxy for upsampling, and where is the parameter reflected in Train_ZSDeconvNet_2D.py? Another issue is that I found when I define different PSF matrix size and keep other PSF parameters the same, the deconvolution effect of the trainning model is very different, should there be limitations on the size setting of PSF?  Looking forward to your reply to the above questions，thanks! 
fig1 is the PSF parameters we set. Fig2 is deconvoluted_50000iter  result with size 527x527 PSF. Fig3 is deconvoluted_50000iter  result with size 851x851 PSF. Fig4 is the parameters in Train_ZSDeconvNet_2D.py we used. 
![psf设置](https://github.com/user-attachments/assets/23a56026-f3b4-42e7-8ce4-a3cb68f6401b)
![psf527_527](https://github.com/user-attachments/assets/7a527c36-c0a5-4ac7-bdfe-16c0eef7ed6d)
![psf851_851](https://github.com/user-attachments/assets/9de6de00-e9de-47de-a5d7-93c2cd4d4ead)
![trainModelParSet](https://github.com/user-attachments/assets/acc714be-7a36-4de8-ad3e-335b29db8522)

**Comment by TristaZeng** [comment]
Please check the simulated PSF for any difference except for the size. If the size is large enough, further increasing the PSF size is just padding more zeros on the edges.
If you set `upsample_flag=1` in `Train_ZSDeconvNet_2D.py`, `dx` and `dy` should be 1/2 of the raw data lateral sampling distance. Because they are actually the lateral sampling distance of network output.

**Comment by Xuesm** [comment]
Thanks，I find related description about data in train_demo_2D.sh. I will try again. 
And I did use simulated PSF with the same parameters except for the size. I will check again. thank you very much!

**Comment by LQsssa** [comment]
How to adjust PSF parameters to avoid artifacts?
